scripts/ci/libraries/_push_pull_remove_images.sh

```diff
@@ -294,6 +302,46 @@ function push_pull_remove_images::wait_for_github_registry_image() {
     verbosity::print_info "Found ${image_name_in_github_registry}:${image_tag_in_github_registry} image"
 }
 
+
+# waits for an image to be available in GitHub Container Registry
+function push_pull_remove_images::wait_for_image_in_github_container_registry() {
+    local image_name_in_github_registry="${1}"
+    local image_tag_in_github_registry=${2}
+
+    local image_to_wait_for="${GITHUB_REGISTRY}/${GITHUB_REPOSITORY}-${image_name_in_github_registry}:${image_tag_in_github_registry}"
+    echo
+    echo "Waiting for ${GITHUB_REGISTRY}/${GITHUB_REPOSITORY}-${image_name_in_github_registry}:${image_tag_in_github_registry} image"
+    echo
+    set +e
+    while true; do
+        docker manifest inspect "${image_to_wait_for}"
```

Comment:
FYI: GCR supports unauthenticated `docker manifest inspect` for public images, which is a big PLUS.